### PR TITLE
chore(api): return root data on removePath

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: packages/buckets/src/api/index.ts [hunk 1]
```diff
@@ -195,6 +198,23 @@ function fromPbArchive(item: _Archive.AsObject): Archive {
   }
 }
 
+/**
+ * Ensures that a Root | string | undefined is converted into a string
```
Comment (andrewxhill) [inline packages/buckets/src/api/index.ts:202]: consolidate this step since used in two places now

File: packages/buckets/src/api/index.ts [hunk 2]
```diff
@@ -745,16 +749,20 @@ export async function bucketsRemovePath(
...
   req.setPath(path)
-  if (root) req.setRoot(root)
-  await api.unary(APIService.RemovePath, req, ctx)
-  return
+  const root = await ensureRootString(api, key, opts?.root, ctx)
+  req.setRoot(root)
+  const res: _RemovePathResponse = await api.unary(APIService.RemovePath, req, ctx)
+  return {
+    pinned: res.getPinned(),
+    root: fromPbRootObjectNullable(res.getRoot())
```
Comment (andrewxhill) [inline packages/buckets/src/api/index.ts:764]: what i needed to get at

File: packages/buckets/src/buckets.spec.ts [hunk 3]
```diff
@@ -264,6 +265,13 @@ describe('Buckets...', function () {
       }
       list = await client.listPath(rootKey, '')
       expect(list.item?.items).to.have.length(2) // Includes .textileseed
+
+      try {
+        await client.removePath(rootKey, 'path', {root: wrongRoot})
+        throw wrongError
+      } catch (err) {
+        expect(err.message).to.equal('update is non-fast-forward')
```
Comment (andrewxhill) [inline packages/buckets/src/buckets.spec.ts:273]: the proof

Comment (carsonfarmer) [reply packages/buckets/src/buckets.spec.ts:273]: yeh nice test, good to have this in here.

**Review by carsonfarmer — APPROVED** [review]
Nice catching here @andrewxhill.

**Pull request merged**